### PR TITLE
[MM-34306] Migrate 'components/admin_console/brand_image_setting' and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by majidsajadi**
#### Summary

migrated `brand_image_setting` to typescript.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/17244

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by hmhealey — APPROVED** [review]
Thanks for the your continuing help with this migration! This PR looks good to me.

**Comment by hmhealey** [comment]
@majidsajadi Would you be able to fix the merge conflict as well as the test? It looks like there's a couple test cases still using the old import path for the file.

**Review by hahmadia — APPROVED** [review]

**Comment by hahmadia** [comment]
Hey! Could you fix the test and merge conflicts so we can pass onto QA for testing? Thanks

**Comment by majidsajadi** [comment]
@hahmadia merged and resolved.

File: components/admin_console/brand_image_setting/brand_image_setting.tsx [hunk 1]
```diff
@@ -16,41 +15,54 @@ import OverlayTrigger from 'components/overlay_trigger';
...
+    brandImage?: Blob;
+    brandImageExists: boolean;
+    brandImageTimestamp: number;
+    error: string;
+};
+
+export default class BrandImageSetting extends React.PureComponent<
+Props,
+State
+> {
```
Comment (saturninoabril) [inline components/admin_console/brand_image_setting/brand_image_setting.tsx:57]: nit: Please update as one-liner, thanks
```suggestion
export default class BrandImageSetting extends React.PureComponent<Props, State> {
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @majidsajadi! All looks good to me, except for one request then this is ready to merge.

**Review by saturninoabril — APPROVED** [review]
Thanks @majidsajadi!

**Pull request merged**